utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/client/ConfluentSchemaRegistryClient.scala

```diff
@@ -59,6 +62,22 @@ class DefaultConfluentSchemaRegistryClient(override val client: CSchemaRegistryC
       client.getAllVersions(subject).asScala.toList
     }
 
+  override def getSchemaById(id: Int): SchemaWithMetadata = {
+    //todo schemaCache is subject-version keyed, how to get it by id
+    val rawSchema = client.getSchemaById(id)
+    withExtraSchemaTypes(id, rawSchema)
+  }
+
+  private def withExtraSchemaTypes(id: Int, rawSchema: ParsedSchema) = {
```

Comment:
I wonder, maybe let's put it in SchemaWithMetadata.apply? So we have schema dispatch logic in one pllace?

Reply:
done